blackbird_python/blackbird/tests/test_listener.py

```diff
@@ -832,6 +832,24 @@ def test_for_loop(self, modes, parse_input_mocked_metadata):
             {'op': 'MeasureFock', 'args': [], 'kwargs': {}, 'modes': [modes[1]]},
             {'op': 'MeasureFock', 'args': [], 'kwargs': {}, 'modes': [modes[2]]}
         ]
+        
+    def test_for_bool(self, parse_input_mocked_metadata):
+        """Test that a for-loop over a list containing bools is parsed correctly"""
+        bb = parse_input_mocked_metadata(
+            "for bool b in [True, False]\n\tMeasureFock() | 0"
+        )
+        assert np.all(
+            bb._forvar["b"] == np.array([True, False])
+        )
+        
+    def test_for_str(self, parse_input_mocked_metadata):
+        """Test that a for-loop over a list containing strings is parsed correctly"""
+        bb = parse_input_mocked_metadata(
+            'for str s in ["one", "two"]\n\tMeasureFock() | 0'
+        )
+        assert np.all(
+            bb._forvar["s"] == np.array(["one", "two"])
+        )
```

Comment:
Nice tests, although, should they also be testing for the case where the boolean for loop variable is used within the loop?

Reply:
Updated the test to use the bool inside the loop. Not sure if there's another way of testing it. 🤔